src/action-ros-ci.ts

```diff
@@ -183,10 +183,17 @@ async function run() {
 			options
 		);
 
+		// Let Eloquent be the default distro used for rosdep
+		let rosdepRosdistro = "eloquent";
+		// If sourcing a binary installation, then we should use the last overlayed distro for rosdep
+		if (sourceRosBinaryInstallation) {
+			rosdepRosdistro = sourceRosBinaryInstallationList[sourceRosBinaryInstallationList.length - 1];
```

Comment:
Let's add an explicit argument here. Selecting the last one seems a bit random, and will break with ROS 1 / ROS 2 mixed env we need to support.

Reply:
Step 1: add an arg called `rosdep-rosdistro` in `action.yml`, default it to `eloquent`.
Step 2: use the newly added argument in the TS file.

Reply:
What if we install dependencies for all sourced distros instead? I guess that should not break any use case (and would fix my issue).

If we add a new argument, I think it should accept one or more distros, so that we can install dependencies from multiple distros accordingly. If we go this far then the information seems redundant with the existing `source-ros-binary-installation` arg. It's only redundant assuming we should always install dependencies for each distro, if this assumption is wrong then adding a new argument makes sense. But, I expect most of the time, if someone wants to source multiple distros, they probably want to install dependencies for both, e.g.

```yaml
source-ros-binary-installation: melodic dashing
rosdep-rosdistro: melodic dashing
```

Reply:
+1 to everything you said. I'd be OK starting with just installing deps for all distros we source, and then introduce the flag you described as an optimization for speed/advanced users. Or do both in this PR. Thanks for the suggestion!

Reply:
I've add logic to install deps for every sourced distro: f687ed0